log-analyzer/features/org.wso2.carbon.la.apim.feature/src/main/resources/gadgets/LogAPIMArtifactDeleted/js/VizGrammar.js

```diff
@@ -0,0 +1,2300 @@
+    if (this.config.maxLength != -1 && this.config.maxLength <  (this.view.data(this.config.title).values().length + data.length)) {
+        var removeFunction = (function(d) {
+            return d[this.metadata.names[this.config.x]] == oldData;
+        }).bind(this);
+
+        for (i = 0; i < data.length; i++) {
+            var oldData = this.view.data(this.config.title).values()[i][this.metadata.names[this.config.x]];
+            this.view.data(this.config.title).remove(removeFunction);
+        }
+    }
+
+    this.view.data(this.config.title).insert(data);
+    this.view.update();
+};
+
+line.prototype.getSpec = function() {
+    return this.spec;
+};
+
+function getLineMark(config, metadata){
+    var mark;
+    if (config.color != -1) {
+        mark =  {
+            "type": "group",
+            "from": {
+                "data":  config.title,
+                "transform": [{"type": "facet", "groupby": [metadata.names[config.color]]}]
+            },
+            "marks": [
+                {
+                    "type": "line",
+                    "properties": {
+                        "update": {
+                            "x": {"scale": "x", "field": metadata.names[config.x]},
+                            "y": {"scale": "y", "field": metadata.names[config.y]},
+                            "stroke": {"scale": "color", "field": metadata.names[config.color]},
+                            "strokeWidth": {"value": 2},
+                            "strokeOpacity": {"value": 1}
+                        },
+                        "hover": {
+                            "strokeOpacity": {"value": 0.5}
+                        }
+                    }
+                }
+            ]
+        };
+    } else {
+        mark = {
+            "type": "line",
+            "from": {"data": config.title},
+            "properties": {
+                "update": {
+
+                    "x": {"scale": "x", "field": metadata.names[config.x]},
+                    "y": {"scale": "y", "field": metadata.names[config.y]},
+                    "stroke": { "value": config.markColor},
+                    "strokeWidth": {"value": 2},
+                    "strokeOpacity": {"value": 1}
+                },
+                "hover": {
+                    "strokeOpacity": {"value": 0.5}
+                }
+            }
+        };
+    }
+
+    return mark;
+}
+
+;var map = function(dataTable, config) {
+
+    this.metadata = dataTable[0].metadata;
+    var marks ;
+    var signals ;
+    var predicates = [];
+    var legends = [];
+    this.spec = {};
+    var geoInfoJson ;
+
+    geoInfoJson = loadGeoMapCodes(config.helperUrl);
+    config = checkConfig(config, this.metadata);
+    this.config = config;
+    this.config.geoInfoJson = geoInfoJson;
+
+    for (i = 0; i < dataTable[0].values.length; i++) {
+        for (var key in dataTable[0].values[i]) {
+            if(key == dataTable[0].metadata.names[config.x]){
```

Comment:
Hi Chandana,
We have replace the Vizgrammer.js with Vizgrammer.min.js and relevant commits are in

https://github.com/wso2/shared-analytics/pull/126/files#diff-6ac780d61b2b6d470c8ff04bdeed3bcdR52

Thanks,
SajithD